### ISSUE TITLE
Enhancement: [member-ordering] Port requiredFirst from eslint-plugin-typescript-sort-keys

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#5596]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/member-ordering/

### Description

[eslint-plugin-typescript-sort-keys](https://github.com/infctr/eslint-plugin-typescript-sort-keys) supports a `requiredFirst` option that enforces required members come before optional. Can we add that to `@typescript-eslint/member-ordering`, to bring it up to parity?

Porting discussion from https://github.com/infctr/eslint-plugin-typescript-sort-keys/issues/50#issuecomment-1108934841

### Fail

```typescript
interface U {
  d: T
  c?: T
  b?: T
  a: T
}
```

### Pass

```typescript
interface U {
  a: T
  d: T
  b?: T
  c?: T
}
```

### Additional Info

Examples taken from https://github.com/infctr/eslint-plugin-typescript-sort-keys/blob/master/docs/rules/interface.md.

**Comment by asdf93074** [comment]
Hey, I can give it a go. I assume if `requiredFirst` is true then the properties should be sorted with all optional ones at the end and then we should apply whatever grouping and sort order they asked for?

**Comment by JoshuaKGoldberg** [comment]
That sounds right to me! 👍

**Comment by JoshuaKGoldberg** [comment]
:wave: @asdf93074 just checking in, do you still have time & interest for this issue?

**Comment by asdf93074** [comment]
Hey, apologies. I've been away from this codebase for a while now. I did write a basic solution for it but I still need to handle most of the cases that crop up. If anyone else wants to pick this up, feel free.

**Comment by asdf93074** [comment]
@JoshuaKGoldberg I put in some time and opened a draft PR for it. Just wondering, how extensive should the test cases be here? Does it need to add tests for all member groups x all 4 member name sorts?

**Comment by JoshuaKGoldberg** [comment]
Hmm, good question. I don't think these need to cover all member groups x member name sorts. https://github.com/typescript-eslint/typescript-eslint/blob/8c3e9d4315c9367fa9cdc86103bbcf954622b57e/packages/eslint-plugin/tests/rules/member-ordering/member-ordering-required-first.test.ts looks reasonable to me. Thanks for asking!
